Fix text overflow in category navigation menus

Rafikooo opened the pull request:
## Summary

Fixes text overflow issues when category names are too long in navigation menus across all resolutions.

**Implementation:**
- Created reusable SCSS mixins for text-overflow handling (`mixins/_text-truncate.scss`)
- Refactored all text-overflow CSS to use mixins for better maintainability
- Improved breadcrumb `max-width` to use responsive `min(300px, 30vw)` instead of fixed `300px`

**Affected areas:**
- ShopBundle: navigation dropdown (all resolutions), mobile offcanvas menu, category page headers
- AdminBundle: taxon tree sidebar, breadcrumbs

Text is now properly truncated with ellipsis (...) and users can see full names via tooltip on hover.

Originally reported issue:
<img width="818" height="388" alt="image" src="https://github.com/user-attachments/assets/a184a860-ce03-4ece-bdfe-9292d1ed375a" />

I enhanced a few other places as well:

Before:
<img width="1438" height="891" alt="image" src="https://github.com/user-attachments/assets/e4b8dced-5e0e-4fdf-bb46-faf4ecb2081b" />
After:
<img width="1481" height="916" alt="image" src="https://github.com/user-attachments/assets/dfd09dc5-f187-4abd-901d-2c30ea979017" />

---
Before:
<img width="807" height="772" alt="image" src="https://github.com/user-attachments/assets/d781ae3a-0118-409a-8b39-24def88af256" />
After:
<img width="859" height="752" alt="image" src="https://github.com/user-attachments/assets/a4fdf0f0-0cb5-4a09-9d0f-099fb216e591" />

Before:
<img width="1282" height="808" alt="image" src="https://github.com/user-attachments/assets/2e31a2aa-a3a5-45f0-a424-b0c0d731f53f" />
After:
<img width="1376" height="851" alt="image" src="https://github.com/user-attachments/assets/d5477492-643a-4dde-8466-2c1ec2632858" />


Other:
<img width="890" height="874" alt="image" src="https://github.com/user-attachments/assets/b5bc6488-c449-4d21-91cd-173d23e46b9c" />
<img width="312" height="513" alt="image" src="https://github.com/user-attachments/assets/1ef36b3d-691b-44d3-b26f-102bad5ecda7" />

<img width="1446" height="602" alt="image" src="https://github.com/user-attachments/assets/880cee99-fb07-43cb-94a3-ff4f3da6e56f" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added text truncation and responsive text-wrapping for long text in breadcrumbs, navigation menus, and tree views.
  * Added tooltip support for navigation items and breadcrumb elements.

* **Bug Fixes**
  * Improved text overflow handling across UI components for better readability.
  * Enhanced HTML escaping for improved security.
  * Icon error handling now surfaces missing icons as exceptions.

* **Style**
  * Increased dropdown menu width and improved responsive behavior across breakpoints.
  * Optimized text display in product headers and offcanvas navigation.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

PiotrTulacz commented on src/Sylius/Bundle/ShopBundle/Resources/assets/styles/_base.scss:
```diff
@@ -20,6 +20,11 @@ a {
+h1 {
+    overflow-wrap: break-word;
+    word-break: break-word;
+}
```
I’m wondering if it’s the right call to add this rule globally. Wouldn’t it be safer to create a dedicated SCSS utility or apply it only in the specific spot that needs it? The behavior is useful, but overriding every h1 might
break existing projects/themes that rely on default heading behavior.

PiotrTulacz commented on src/Sylius/Bundle/ShopBundle/Resources/assets/styles/_nav.scss:
```diff
@@ -2,17 +2,27 @@
+
+  @media (min-width: 992px) and (max-width: 1199px) {
+    .dropdown-menu {
+      max-width: 30rem;
```
“Can we also nest this into .dropdown-menu?

PiotrTulacz commented on src/Sylius/Bundle/ShopBundle/Resources/assets/styles/_offcanvas.scss:
```diff
@@ -10,4 +10,36 @@
+        .nav-link {
+            max-width: 100%;
+            box-sizing: border-box;
+            @include text-truncate-ellipsis;
```
Cool, this behavior will hide any overflowing text.

Rafikooo committed:
Fix text overflow in category navigation menus

- Add reusable SCSS mixins for text truncation (ellipsis, flex)
- Fix long category names in navbar dropdown (desktop: 1-line ellipsis, mobile: 2-line wrap)
- Fix taxon tree overflow in admin sidebar using flex truncation
- Fix XSS vulnerability in taxon tree by escaping HTML
- Remove global h1 overflow rule, use .text-break class instead
- Ensure offcanvas dropdown menu has no border on mobile/tablet
